wasm/derive/src/validator/entrypoint.rs

```diff
@@ -84,6 +78,35 @@ impl crate::params::ConstructArg for ParamType {
     }
 }
 
+fn infer_args(sig: &syn::Signature) -> Punctuated<syn::Expr, syn::token::Comma> {
+    let mut args = vec![];
+
+    for input in &sig.inputs {
+        let syn::FnArg::Typed(syn::PatType {ty, ..}) = input else {
+            panic!("Receiver types are not allowed")
+        };
+
+        args.push(if is_authority(ty) {
+            ParamType::Authority
+        } else if is_transaction(ty) {
+            ParamType::Transaction
+        } else if is_instruction(ty) {
+            ParamType::Instruction
+        } else if is_query(ty) {
+            ParamType::Query
+        } else if is_expression(ty) {
+            ParamType::Expression
+        } else {
+            abort!(
+                input,
+                "Unable to infer function parameter type. Use #[entrypoint(params = \"[...]\")]"
+            )
+        });
+    }
+
+    args.into_iter().map(|arg| arg.construct_arg()).collect()
```

Comment:
It seems we can avoid a vector allocation here.
```suggestion
    sig.inputs
        .iter()
        .map(|input| {
            let syn::FnArg::Typed(syn::PatType {ty, ..}) = input else {
                panic!("Receiver types are not allowed")
            };

            let param_type = if is_authority(ty) {
                ParamType::Authority
            } else if is_transaction(ty) {
                ParamType::Transaction
            } else if is_instruction(ty) {
                ParamType::Instruction
            } else if is_query(ty) {
                ParamType::Query
            } else if is_expression(ty) {
                ParamType::Expression
            } else {
                abort!(
                input,
                "Unable to infer function parameter type. Use #[entrypoint(params = \"[...]\")]"
            )
            };
            param_type.construct_arg()
        })
        .collect()
```